sdk/core/azure-core/inc/azure/core/http/win_http_transport.hpp

```diff
@@ -9,56 +9,38 @@
 #pragma once
 
 #include "azure/core/context.hpp"
+#include "azure/core/nullable.hpp"
+#include "azure/core/url.hpp
+#include "azure/core/internal/unique_handle.hpp"
 #include "azure/core/http/http.hpp"
-#include "azure/core/http/policies/policy.hpp"
 #include "azure/core/http/transport.hpp"
-#include "azure/core/internal/unique_handle.hpp"
-#include "azure/core/platform.hpp"
-
-#if defined(AZ_PLATFORM_WINDOWS)
-#if !defined(WIN32_LEAN_AND_MEAN)
-#define WIN32_LEAN_AND_MEAN
-#endif
-#if !defined(NOMINMAX)
-#define NOMINMAX
-#endif
-#include <windows.h>
-#endif
+#include "azure/core/http/policies/policy.hpp"
 
+#include <string>
 #include <memory>
-#include <type_traits>
 #include <vector>
-#include <wincrypt.h>
-#include <winhttp.h>
 
 namespace Azure { namespace Core {
```

Comment:
If at all possible, remove this from the public header.

Reply:
I'm sorry, I don't understand which line you are referring to, I will probably follow up with you offline.

Reply:
Sorry - I'm proposing a new guideline that we don't define types in the `_internal` or `_detail` namespace (because it adds to customer confusion - are these types in the public header safe to use?). So if it's possible, I'd love to see if the _detail typedefs could possibly be removed.

Reply:
I don't know if it is possible or not, to be honest. And I wouldn't go crazy restructuring things to isolate public and private, but it would be nice to keep the public surface strictly public.

Reply:
I think we (should) have a paragraph in our docs, saying to customers that they should not write `_detail::` and `_internal::`, as well as `#include <azure/**/internal/**>` - that stuff is not meant for their direct consumption and is subject to breaking changes without warning.
Including a public header, basically, will eventually include a header with `_detail` or `_internal` namespace, after which it is available to the customer's IntelliSense, which should move it to the bottom of the list.

So, you are proposing to only optimize the stuff for the customers who read headers directly? Then, in some cases, including this one, it might be possible to extract that surface into a new `azure/**/internal/**` header.

I could do that, however I think it is better to log that as an issue, and do it for the entire SDK as a separate PR.

Please let me know if you still would recommend to do it for this PR for just the `win_http_transport.hpp` file. I do not feel strongly about it.

Reply:
Ok! I moved out the `_detail` definitions.